src/Model/Casts/DateTime/BaseDateTimeCast.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Mongolid\Model\Casts\DateTime;
+
+use DateTimeInterface;
+use MongoDB\BSON\UTCDateTime;
+use MongoDB\BSON\UTCDateTimeInterface;
+use Mongolid\Model\Casts\CasterInterface;
+use Mongolid\Model\Casts\CastInterface;
+
+abstract class BaseDateTimeCast implements CastInterface
+{
+    /**
+     * @param UTCDateTime|null $value
+     * @return DateTimeInterface|null
+     */
+    abstract public function get(mixed $value): DateTimeInterface|null;
```

Comment:
If the return type is set, I think you don't need of the annotation.
```suggestion
     * @param ?UTCDateTime $value
     */
    abstract public function get(mixed $value): ?DateTimeInterface;
```

Also, what do you think about have a specific interface to each type of caster?

Reply:
About a specific interface for each type of caster, it is not necessary to specify it. 

A generic interface is needed in order to make `Caster::resolve()` transparent for clients: you just need to call it and then use `get` or `set` to Cast objects, just like @edipoReboucas also did at the EnumCast implementation: https://github.com/leroy-merlin-br/mongolid/pull/224

Thats the power of declaring a generic interface for this kind of objects. But declaring specific ones would provide zero to no value for this lib.